distributed/shuffle/_scheduler_extension.py

```diff
@@ -227,6 +277,22 @@ def _create_array_rechunk_state(
             participating_workers=output_workers.copy(),
         )
 
+    def _set_restriction(self, ts: TaskState, worker: str) -> None:
+        if "shuffle_original_restrictions" in ts.annotations:
+            # This may occur if multiple barriers share the same output task,
+            # e.g. in a hash join.
+            return
+        ts.annotations["shuffle_original_restrictions"] = ts.worker_restrictions.copy()
```

Comment:
Dummy question, sorry. The problem that you're fixing is tasks occasionally losing annotations. AFAIK this is applicable to all annotations, not just `shuffle` annotations, and your `ShuffleAnnotationChaosPlugin` is only killing`shuffle` annotation. Is there a possibility that in a real-world scenario, this new `shuffle_original_restrictions` will be lost too?

Reply:
Based on what we've seen, the current assumption is that `annotations` get lost _before_ the tasks make it to the scheduler.